From 0e042d322ce6c7a227edccb6c4f0d089cc791054 Mon Sep 17 00:00:00 2001
From: Wang Hui <Hui.Wang@windriver.com>
Date: Wed, 4 Nov 2009 14:11:16 +0800
Subject: [PATCH] musb/otg: Add otg transceiver in Makefile and Kconfig

On Omap3evm platform, the default transceiver for musb otg
is isp1504 rather twl4030, because the 0ohm resistors between musb and
isp1504 is soldered, while the ones between musb and twl4030 is
not populated. Here add isp1504 transceiver in Makefile and Kconfig.

[Reference to upstream commit ID: b8da867 f6d92a0 e8e2ff4, from these
commit, picking usb OTG transceiver related change in Makefile and
Kconfig.]

Signed-off-by: Wang Hui <Hui.Wang@windriver.com>
---
 drivers/Makefile         |    1 +
 drivers/usb/Kconfig      |    2 ++
 drivers/usb/musb/Kconfig |    2 ++
 drivers/usb/otg/Kconfig  |   28 ++++++++++++++++++++++++++++
 drivers/usb/otg/Makefile |   12 ++++++++++++
 5 files changed, 45 insertions(+), 0 deletions(-)
 create mode 100644 drivers/usb/otg/Kconfig
 create mode 100644 drivers/usb/otg/Makefile

diff --git a/drivers/Makefile b/drivers/Makefile
index 39cfe40..a1f8bc7 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -62,6 +62,7 @@ obj-$(CONFIG_MAC)		+= macintosh/
 obj-$(CONFIG_ATA_OVER_ETH)	+= block/aoe/
 obj-$(CONFIG_PARIDE) 		+= block/paride/
 obj-$(CONFIG_TC)		+= tc/
+obj-$(CONFIG_USB_OTG_UTILS)	+= usb/otg/
 obj-$(CONFIG_USB)		+= usb/
 obj-$(CONFIG_USB_MUSB_HDRC)	+= usb/musb/
 obj-$(CONFIG_PCI)		+= usb/
diff --git a/drivers/usb/Kconfig b/drivers/usb/Kconfig
index f2df4f2..9ca5341 100644
--- a/drivers/usb/Kconfig
+++ b/drivers/usb/Kconfig
@@ -149,4 +149,6 @@ source "drivers/usb/atm/Kconfig"
 
 source "drivers/usb/gadget/Kconfig"
 
+source "drivers/usb/otg/Kconfig"
+
 endif # USB_SUPPORT
diff --git a/drivers/usb/musb/Kconfig b/drivers/usb/musb/Kconfig
index 4b9542b..2d279f7 100644
--- a/drivers/usb/musb/Kconfig
+++ b/drivers/usb/musb/Kconfig
@@ -11,6 +11,8 @@ config USB_MUSB_HDRC
 	depends on (USB || USB_GADGET) && HAVE_CLK
 	depends on !SUPERH
 	select TWL4030_USB if MACH_OMAP_3430SDP
+	select NOP_USB_XCEIV if MACH_OMAP3EVM
+	select USB_OTG_UTILS
 	tristate 'Inventra Highspeed Dual Role Controller (TI, ...)'
 	help
 	  Say Y here if your system has a dual role high speed USB
diff --git a/drivers/usb/otg/Kconfig b/drivers/usb/otg/Kconfig
new file mode 100644
index 0000000..12f11f7
--- /dev/null
+++ b/drivers/usb/otg/Kconfig
@@ -0,0 +1,28 @@
+#
+# USB OTG infrastructure may be needed for peripheral-only, host-only,
+# or OTG-capable configurations when OTG transceivers or controllers
+# are used.
+#
+
+comment "OTG and related infrastructure"
+
+config USB_OTG_UTILS
+	bool
+	help
+	  Select this to make sure the build includes objects from
+	  the OTG infrastructure directory.
+
+if USB || USB_GADGET
+
+#
+# USB Transceiver Drivers
+#
+config NOP_USB_XCEIV
+	tristate "NOP USB Transceiver Driver"
+	select USB_OTG_UTILS
+	help
+	 this driver is to be used by all the usb transceiver which are either
+	 built-in with usb ip or which are autonomous and doesn't require any
+	 phy programming such as ISP1x04 etc.
+
+endif # USB || OTG
diff --git a/drivers/usb/otg/Makefile b/drivers/usb/otg/Makefile
new file mode 100644
index 0000000..4ca0419
--- /dev/null
+++ b/drivers/usb/otg/Makefile
@@ -0,0 +1,12 @@
+#
+# OTG infrastructure and transceiver drivers
+#
+
+# infrastructure
+
+# transceiver drivers
+obj-$(CONFIG_NOP_USB_XCEIV)	+= nop-usb-xceiv.o
+
+ccflags-$(CONFIG_USB_DEBUG)	+= -DDEBUG
+ccflags-$(CONFIG_USB_GADGET_DEBUG) += -DDEBUG
+
-- 
1.6.0.3

